src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -2051,84 +2055,107 @@ public String generateSummaryText(int numRecordsRead,
 		return summary;
 	}
 
+	public Collection<BulkLoadParentSaltFormLotDTO> getBulkLoadDTOsByBulkLoadFileId(Long bulkLoadFileId) {
+		// This query pulls back the corp name and bulk load file ids for all parents, saltforms, and lots that are at all linked
+		// to the bulkLoadFileId in question. 
+		EntityManager em = Parent.entityManager();
+        TypedQuery<BulkLoadParentSaltFormLotDTO> q = em.createQuery("SELECT new com.labsynch.labseer.dto.BulkLoadParentSaltFormLotDTO("
```

Comment:
Here's the main query that allows us to skip a lot of repeat work. We're grabbing just the corpName and bulkLoadFile IDs from the parent, saltForm, and lot tables, where any of the three (parent, saltform, or lot) are tied to the current bulkLoadFile that's being assessed for purging.

This means we'll get some repeated values in some columns, for example if a parent has three lots, one from the same file, one from a different file, and one from single registration (bulkLoadFile = null).

In SQL this query equates to:
```
select p.corp_name, p.bulk_load_file, s.corp_name, s.bulk_load_file, l.corp_name, l.bulk_load_file
from parent p
join salt_form s on p.id = s.parent
join lot l on l.salt_form = s.id
where p.bulk_load_file = 2091 or s.bulk_load_file = 2091 or l.bulk_load_file = 2091;
```
where 2091 is an example bulkLoadFileId.

Reply:
Thanks, fixed it.